taskchampion/src/storage/sqlite.rs

```diff
@@ -1,355 +1,346 @@
+            )
+            .optional()?;
+        Ok(version.map(|u| u.0).unwrap_or(DEFAULT_BASE_VERSION))
     }
 
     fn set_base_version(&mut self, version: VersionId) -> anyhow::Result<()> {
-        let uuids_bucket = self.uuids_bucket();
-        let kvtxn = self.kvtxn();
-
-        kvtxn.set(
-            uuids_bucket,
-            BASE_VERSION.into(),
-            Msgpack::to_value_buf(version as Uuid)?,
-        )?;
+        let t = self.get_txn()?;
+        t.execute(
+            "INSERT OR REPLACE INTO sync_meta (key, value) VALUES (?, ?)",
+            params!["base_version", &StoredUuid(version)],
+        )
+        .context("Set base version")?;
         Ok(())
     }
 
     fn operations(&mut self) -> anyhow::Result<Vec<Operation>> {
-        let bucket = self.operations_bucket();
-        let kvtxn = self.kvtxn();
-        let all_ops: Result<Vec<(u64, Operation)>, Error> = kvtxn
-            .read_cursor(bucket)?
-            .iter()
-            .map(|(i, v)| Ok((i.into(), v.inner()?.to_serde())))
-            .collect();
-        let mut all_ops = all_ops?;
-        // sort by key..
-        all_ops.sort_by(|a, b| a.0.cmp(&b.0));
-        // and return the values..
-        Ok(all_ops.iter().map(|(_, v)| v.clone()).collect())
+        let t = self.get_txn()?;
+
+        let mut q = t.prepare("SELECT data FROM operations ORDER BY id ASC")?;
+        let rows = q.query_map([], |r| {
+            let data: Operation = r.get("data")?;
+            Ok(data)
+        })?;
+
+        let mut ret = vec![];
+        for r in rows {
+            ret.push(r?);
+        }
+        Ok(ret)
     }
 
     fn add_operation(&mut self, op: Operation) -> anyhow::Result<()> {
-        let numbers_bucket = self.numbers_bucket();
-        let operations_bucket = self.operations_bucket();
-        let kvtxn = self.kvtxn();
-
-        let next_op = match kvtxn.get(numbers_bucket, NEXT_OPERATION.into()) {
-            Ok(buf) => buf.inner()?.to_serde(),
-            Err(Error::NotFound) => 0,
-            Err(e) => return Err(e.into()),
-        };
-
-        kvtxn.set(
-            operations_bucket,
-            next_op.into(),
-            Msgpack::to_value_buf(op)?,
-        )?;
-        kvtxn.set(
-            numbers_bucket,
-            NEXT_OPERATION.into(),
-            Msgpack::to_value_buf(next_op + 1)?,
-        )?;
+        let t = self.get_txn()?;
+
+        t.execute("INSERT INTO operations (data) VALUES (?)", params![&op])
+            .context("Add operation query")?;
         Ok(())
     }
 
     fn set_operations(&mut self, ops: Vec<Operation>) -> anyhow::Result<()> {
-        let numbers_bucket = self.numbers_bucket();
-        let operations_bucket = self.operations_bucket();
-        let kvtxn = self.kvtxn();
-
-        kvtxn.clear_db(operations_bucket)?;
+        let t = self.get_txn()?;
+        t.execute("DELETE FROM operations", [])
+            .context("Clear all existing operations")?;
```

Comment:
Maybe also reset the indexing? https://www.designcise.com/web/tutorial/how-to-reset-autoincrement-number-sequence-in-sqlite#deleting-the-table-number-sequence

Reply:
Makes sense - done!